Makefile.gcc4

```diff
@@ -0,0 +1,33 @@
+# Makefile for ancient gcc4 (tested on Ubuntu 7.10 with 4.1.2-16ubuntu2)
+# Compiles Kaitai Struct C++ runtime and tests
+
+BUILD_DIR := build
+
+SOURCES := \
+	kaitai/kaitaistream.cpp \
+	tests/unittest.cpp
+
+OBJS := \
+	$(BUILD_DIR)/kaitaistream.o \
+	$(BUILD_DIR)/unittest.o
+
+DEFINES := -DKS_STR_ENCODING_ICONV -DGTEST_NANO
+
+LFLAGS :=
+LIBS :=
+
+INCLUDES := -I.
+
+all: $(BUILD_DIR) $(BUILD_DIR)/unittest
+
+$(BUILD_DIR):
+	mkdir -p $(BUILD_DIR)
+
+$(BUILD_DIR)/unittest: $(OBJS)
+	$(CXX) $(CXXFLAGS) -o build/unittest $(OBJS) $(LFLAGS) $(LIBS)
```

Comment:
Technically, the standard names (used also in implicit rules of GNU make) for what you call _`LFLAGS`_ and _`LIBS`_ would be `LDFLAGS` and `LDLIBS` (see https://www.gnu.org/software/make/manual/html_node/Catalogue-of-Rules.html#index-linking_002c-predefined-rule-for) - see also https://www.gnu.org/software/make/manual/html_node/Implicit-Variables.html#index-LDFLAGS. Of course, such a convention is not that important, but I guess it's good to follow.

Reply:
Thanks for the catch, just fixed!